test/ng/directive/ngModelOptionsSpec.js

```diff
@@ -480,15 +480,37 @@ describe('ngModelOptions', function() {
         expect($rootScope.checkbox).toBeUndefined();
         $timeout.flush(4000);
         expect($rootScope.name).toEqual('a');
+
         helper.changeInputValueTo('b');
         browserTrigger(inputElm, 'blur');
         $timeout.flush(4000);
         expect($rootScope.name).toEqual('a');
         $timeout.flush(2000);
         expect($rootScope.name).toEqual('b');
+
+        helper.changeInputValueTo('c');
+        browserTrigger(helper.inputElm, 'mouseup');
+        // counter-intuitively `default` in `debounce` is a catch-all
+        expect($rootScope.name).toEqual('b');
+        $timeout.flush(10000);
+        expect($rootScope.name).toEqual('c');
       });
 
 
+      it('should trigger immediately for the default event if not listed in the debounce list',
```

Comment:
The trouble with the test as you suggest is that the `default` event is triggered by `helper.changeInputValueTo('b')` in any case. And so this would pass even if the `foo` event had a weird truthy timeout.